### DISCUSSION TITLE
Processes not visible in WIndows Task Manager after update to WSL 2

### DISCUSSION THREAD

**Discussion by Baterka** [microsoft/WSL#5828] [General]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.450]
Your Distribution version: Debian 10
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.104-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Feb 19 06:37:35 UTC 2020 (It is WSL 2)
```

# Steps to reproduce

Start any service in WSL 2 (I tested it with MariaDB and ssh-server)

#  Expected behavior

Process should be visible in Windows Task Manager as it was in WSL 1.

# Actual behavior

There is just one process named `Vmmem` which is probably the VM.

**Comment by therealkenc** [comment]
This is by-design. WSL2 processes are not Windows processes (unlike in WSL1).

**Comment by Baterka** [comment]
Any way to have that feature back? My GUI interface for start/stop/restart services depends on it.

**Comment by michaelkolber** [comment]
I feel like this is related, so I'll leave it here: if WSL2 processes were to be shown in Task Manager, could they be grouped together the way Windows Explorer is? That way we can easily get a sense of how much a distro is using overall, and also dig into individual processes.

**Comment by therealkenc** [comment]
Sorry this was totally my bad re-opening (my eyes crossed tagging network issues, guh).

> Any way to have that feature back? 

No, they aren't windows processes anymore. You need to communicate with the running distribution's process status. For example spawning "`wsl.exe -d <distro> ps ax`" and parsing the output, or something more elaborate. Moving to discussion rather than closing outright again. Again apologies to the confused tagging.

**Comment by jfehrle** [comment]
They may not be Windows processes any more, but nonetheless it would be quite helpful to see that info since Task Manager gives an overall view of performance.  Vmmem can only give a partial view.

**Comment by Biswa96** [comment]
Open one WSL window with htop. Who need Task Manager? 🤓

**Reply by jfehrle** [reply]
Nice, but only shows the WSL processes.

**Comment by therealkenc** [comment]
There is another feature request for this at #6881. For those interested like and subscribe that ticket.
